src/DonationForms/Listeners/StoreCustomFields.php

```diff
@@ -21,20 +25,56 @@ public function __invoke(DonationForm $form, Donation $donation, array $customFi
     {
         $form->schema()->walkFields(
             function (Field $field) use ($customFields, $donation) {
-                if (!array_key_exists($field->getName(), $customFields)) {
+                $fieldName = $field->getName();
+                $fieldType = $field->getType();
+
+                if (!array_key_exists($fieldName, $customFields)) {
                     return;
                 }
 
-                $value = $customFields[$field->getName()];
-
-                if ($field->shouldStoreAsDonorMeta()) {
-                    // save as donor meta
-                    give()->donor_meta->update_meta($donation->donorId, $field->getName(), $value);
+                /** @var File $field */
+                if (($fieldType === Types::FILE) && isset($_FILES[$fieldName])) {
+                    //TODO: let file field provide this storage logic
+                    $this->handleFileUpload($field, $donation);
```

Comment:
This is happening _after_ validation, right?

Reply:
yup! that's one more advantage to keeping everything in one flow because the fields will run through the same validation - we just need to add a file rule